examples/demo-app/src/actions.js

```diff
@@ -62,45 +71,167 @@ export function setLoadingMapStatus(isMapLoading) {
+    if (options.keplergl) {
+      dispatch(loadRemoteMap({dataUrl: options.keplergl}));
+    } else {
+      dispatch(loadRemoteSampleMap(options));
+    }
+
     dispatch(setLoadingMapStatus(true));
   };
 }
 
-function loadMapCallback(dispatch, error, result, sample, config) {
-  if (error) {
-    Console.warn(`Error loading datafile ${sample.dataUrl}`);
-    // dispatch(ERROR)
-  } else {
-    dispatch(loadResponseFromRemoteFile(result, config, sample));
-    dispatch(toggleModal(null));
+/**
+ * Load remote map with config and data
+ * @param options {configUrl, dataUrl}
+ * @returns {Function}
+ */
+function loadRemoteSampleMap(options) {
+  return (dispatch) => {
+    // Load configuration first
+    const {configUrl, dataUrl} = options;
+
+    Promise
+      .all([loadRemoteConfig(configUrl), loadRemoteData(dataUrl)])
+      .then(
+        ([config, data]) => loadMapCallback(dispatch, null, data, options, config),
+        error => {
+          if (error) {
+            Console.warn(`Error loading map data`);
+            Console.warn(options);
+            Console.warn(error)
+          }
+        }
+      );
   }
 }
 
-function loadRemoteMap(sample) {
-  return (dispatch) => {
-    // Load configuration first
-    requestJson(sample.configUrl, (confError, config) => {
-      if (confError) {
-        Console.warn(`Error loading config file ${sample.configUrl}`);
-        // dispatch(error)
-        return;
-      }
+/**
+ *
+ * @param url
+ * @returns {Promise<any>}
+ */
+function loadRemoteConfig(url) {
+  if (!url) {
+    // TODO: we should return reject with an appropriate error
+    return Promise.resolve(null)
+  }
 
-      let requestMethod = requestText;
-      if (sample.dataUrl.includes('.json') || sample.dataUrl.includes('.geojson')) {
-        requestMethod = requestJson;
-      }
+  return new Promise((resolve, reject) => {
+    requestJson(url, (error, config) => {
+      error ? reject(error) : resolve(config)
+    })
+  })
+}
+
+/**
+ *
+ * @param url to fetch data from (csv, json, geojson)
+ * @returns {Promise<any>}
+ */
+function loadRemoteData(url) {
+  if (!url) {
+    // TODO: we should return reject with an appropriate error
+    return Promise.resolve(null)
+  }
+
+  let requestMethod = requestText;
+  if (url.includes('.json') || url.includes('.geojson')) {
+    requestMethod = requestJson;
+  }
 
-      // Load data
-      requestMethod(sample.dataUrl, (dataError, result) => {
-        loadMapCallback(dispatch, dataError, result, sample, config);
-      });
+  // Load data
+  return new Promise((resolve, reject) => {
+    requestMethod(url, (error, result) => {
+      // TODO: we need to let users know about CORS issues
+      error ? reject(error) : resolve(result)
     })
+  });
+}
+
+/**
+ *
+ * @param dispatch
+ * @param error
+ * @param result
+ * @param options
+ * @param config
+ */
+function loadMapCallback(dispatch, error, result, options, config) {
+  if (error) {
```

Comment:
you can return a `dispatch => {}` function, instead of passing the diaptch around
```
return dispatch => {
    if (error) {...}
    else {
       dispatch(...)
    }
}
```

Reply:
I will review this one. I think before this diff we were using loadMapCallback in multiple locations but now we may not need to encapsulate it anymore